From 0282bce11918ec0dd0c4e7e3bd54e392c2b27a51 Mon Sep 17 00:00:00 2001
Message-Id: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Fri, 10 Apr 2015 12:06:18 +0530
Subject: [PATCH 01/10] FOR_UPSTREAM [VPG]: drm/i915: Setting the device ready
 during the ULPS exit and enter

The patch sets the "device ready" while setting the ULTS exit and enter
state.

Issue: GMINL-8141
Change-Id: Ie0aa64622cadd0664324926eb1dee90c84fcef27
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 5164f69..2a46d8e 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -212,13 +212,15 @@ static void intel_dsi_device_ready(struct intel_encoder *encoder)
 		val |= pipe ? DELAY_180_PHASE_SHIFT_MIPIC :
 				DELAY_180_PHASE_SHIFT_MIPIA;
 
-	intel_dsi_write_dev_rdy_on_A_and_C(encoder, ULPS_STATE_ENTER);
+	intel_dsi_write_dev_rdy_on_A_and_C(encoder,
+			DEVICE_READY | ULPS_STATE_ENTER);
 
 	/* wait for LP state to go 00 */
 	usleep_range(2500, 3000);
 	I915_WRITE(MIPI_PORT_CTRL(0), val | LP_OUTPUT_HOLD);
 	usleep_range(1000, 1500);
-	intel_dsi_write_dev_rdy_on_A_and_C(encoder, ULPS_STATE_EXIT);
+	intel_dsi_write_dev_rdy_on_A_and_C(encoder,
+			DEVICE_READY | ULPS_STATE_EXIT);
 
 	/* wait for LP state to goto 11 */
 	usleep_range(2500, 3000);
-- 
1.7.9.5

